src/js/api/config.js

```diff
@@ -61,9 +61,61 @@ function _adjustDefaultBwEstimate(estimate) {
     return Defaults.bandwidthEstimate;
 }
 
+function _mergeProperty(localizationObj, allOptionsObj, prop) {
+    const propToCopy = localizationObj[prop] || allOptionsObj[prop];
+
+    if (propToCopy) {
+        localizationObj[prop] = propToCopy;
+    }
+}
+
+function _copyToLocalization(allOptions) {
+    const { advertising, related, sharing, abouttext } = allOptions;
+    const localization = Object.assign({}, allOptions.localization);
+
+    if (advertising) {
+        localization.advertising = localization.advertising || {};
+        _mergeProperty(localization.advertising, advertising, 'admessage');
+        _mergeProperty(localization.advertising, advertising, 'cuetext');
+        _mergeProperty(localization.advertising, advertising, 'loadingAd');
+        _mergeProperty(localization.advertising, advertising, 'podmessage');
+        _mergeProperty(localization.advertising, advertising, 'skipmessage');
+        _mergeProperty(localization.advertising, advertising, 'skiptext');
+    }
+
+    if (related) {
+        if (typeof localization.related === 'string') {
```

Comment:
These if statements shouldn't be nested, because you can have `localization.related === 'string'` without a related block
```
        if (typeof localization.related === 'string') {
            localization.related = {
                heading: localization.related
            };
        } else {
            localization.related = localization.related || {};
        }

if (related) {
        _mergeProperty(localization.related, related, 'autoplaymessage');
}
```